packages/syft/src/syft/serde/json_serde.py

```diff
@@ -0,0 +1,457 @@
+# stdlib
+import base64
+from collections.abc import Callable
+from dataclasses import dataclass
+from enum import Enum
+import json
+import typing
+from typing import Any
+from typing import Generic
+from typing import TypeVar
+from typing import Union
+from typing import get_args
+from typing import get_origin
+
+# third party
+import pydantic
+
+# syft absolute
+import syft as sy
+
+# relative
+from ..server.credentials import SyftSigningKey
+from ..server.credentials import SyftVerifyKey
+from ..types.datetime import DateTime
+from ..types.errors import SyftException
+from ..types.syft_object import BaseDateTime
+from ..types.syft_object_registry import SyftObjectRegistry
+from ..types.uid import LineageID
+from ..types.uid import UID
+from .recursive import DEFAULT_EXCLUDE_ATTRS
+
+T = TypeVar("T")
+
+JSON_CANONICAL_NAME_FIELD = "__canonical_name__"
+JSON_VERSION_FIELD = "__version__"
+JSON_DATA_FIELD = "data"
+
+JsonPrimitive = str | int | float | bool | None
+Json = JsonPrimitive | list["Json"] | dict[str, "Json"]
+
+
+class JSONSerdeError(SyftException):
+    pass
+
+
+@dataclass
+class JSONSerde(Generic[T]):
+    klass: type[T]
+    serialize_fn: Callable[[T], Json] | None = None
```

Comment:
I think this is a bit clunky to read, can we not just have a type check in the register function that checks if its a default json type and then passes a no-op into this?